cluster-autoscaler/expander/factory/chain.go

```diff
@@ -0,0 +1,27 @@
+package factory
+
+import (
+	"k8s.io/autoscaler/cluster-autoscaler/expander"
+
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+type chainStrategy struct {
+	filters  []expander.Filter
+	fallback expander.Strategy
+}
+
+func newChainStrategy(filters []expander.Filter, fallback expander.Strategy) expander.Strategy {
+	return &chainStrategy{
+		filters:  filters,
+		fallback: fallback,
+	}
+}
+
+func (c *chainStrategy) BestOption(options []expander.Option, nodeInfo map[string]*schedulerframework.NodeInfo) *expander.Option {
+	filteredOptions := options
+	for _, filter := range c.filters {
+		filteredOptions = filter.BestOptions(filteredOptions, nodeInfo)
```

Comment:
There is non-trivial logic behind BestOptions call sometime, so maybe worth returning earlier if there's just one option left?

Reply:
+1, will do

Reply:
done